enterprise/backend/src/metabase_enterprise/sandbox/query_processor/middleware/row_level_restrictions.clj

```diff
@@ -365,20 +365,22 @@
 (defn- merge-metadata
   "Merge column metadata from the non-sandboxed version of the query into the sandboxed results `metadata`. This way the
   final results metadata coming back matches what we'd get if the query was not running in a sandbox."
-  [original-metadata metadata]
+  [{::keys [original-metadata] :as query} metadata]
   (letfn [(merge-cols [cols]
             (let [col-name->expected-col (m/index-by :name original-metadata)]
               (for [col cols]
                 (merge
                  col
                  (get col-name->expected-col (:name col))))))]
-    (update metadata :cols merge-cols)))
+    (-> metadata
+        (update :cols merge-cols)
+        (assoc :is_sandboxed (some? (get-in query [::qp.perms/perms :gtaps]))))))
 
 (defenterprise merge-sandboxing-metadata
   "Post-processing middleware. Merges in column metadata from the original, unsandboxed version of the query."
   :feature :sandboxes
-  [{::keys [original-metadata]} rff]
+  [{::keys [original-metadata] :as query} rff]
   (if original-metadata
```

Comment:
I'm not sure what cases `original-metadata` would be `nil`, but if it is, we still want to pass the `:is_sandboxed` flag in the new metadata. Could you refactor this so that `is_sandboxed` is always included?